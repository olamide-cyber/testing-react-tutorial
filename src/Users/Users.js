import React, { useState, useEffect } from 'react';

const Users = () => {
    const [data, setData] = useState([]);

    useEffect(() => {
        const usersResponse = fetch('https://jsonplaceholder.typicode.com/users');
        usersResponse.then(response => {
            if (!response.ok) {
                alert('Unable to fetch usernames, Please check your internet connection and try again.')
            }
            return response.json();
        }).then(jsonResponse => setData(jsonResponse))
    }, []);

    return (
        <>
            {
                data.length === 0 ?  <h1>Loading usernames, please wait...</h1> : (
                    <>
                        <h1>Usernames</h1>
                        <ul>
                            {
                                data.map(user => {
                                    return <li key={user.name}>{user.name}<span className='email'>{user.email}</span></li>
                                })
                            }
                        </ul>
                    </>
                )
            } 
        </>
    );
}

export default Users;