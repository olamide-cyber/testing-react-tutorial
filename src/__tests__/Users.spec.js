import React from 'react';
import renderer from 'react-test-renderer';
import Users from '../Users/Users';

describe('User component snapshot', () => {
  it('matches the component snapshot', () => {
    const tree = renderer.create(<Users />);
    expect(tree).toMatchSnapshot()
  });
});















// import React from 'react';
// import { act } from "react-dom/test-utils";
// import { render, unmountComponentAtNode } from 'react-dom';
// import Users from '../Users/Users';

// let container;

// beforeEach(() => {
//   container = document.createElement("div");
//   document.body.appendChild(container);
// });

// afterEach(() => {
//   unmountComponentAtNode(container);
//   container.remove();
//   container = null;
// });

// describe('User Component', () => {
//     test('It should render a list of users', () => {
//         const fakeResponse = [
//           {name: 'John Doe'},
//           {name: 'Kevin Mitnik'}
//         ]
        
//         jest.spyOn(window, 'fetch').mockImplementation(() => {
//           const fetchResponse = {
//             json: () => Promise.resolve(fakeResponse)
//           };
//           return Promise.resolve(fetchResponse)
//         })

//          act(() => {
//             render(<Users />, container);
//         })
//         expect(container.textContent).toBe('John DoeKevin Mitnick');

//         window.fetch.mockRestore()
//     })
// })