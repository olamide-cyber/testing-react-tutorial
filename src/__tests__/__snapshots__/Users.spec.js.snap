// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component snapshot matches the component snapshot 1`] = `
<h1>
  Loading usernames, please wait...
</h1>
`;
